opslevel/resource_opslevel_team_membership.go

```diff
@@ -0,0 +1,180 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go"
+)
+
+func resourceTeamMembership() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage a team's user membership",
+		Create:      wrap(resourceTeamMembershipCreate),
+		Read:        wrap(resourceTeamMembershipRead),
+		Update:      wrap(resourceTeamMembershipUpdate),
+		Delete:      wrap(resourceTeamMembershipDelete),
+		Importer: &schema.ResourceImporter{
+			State: resourceTeamMembershipImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"team_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"members": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceTeamMembershipCreate(d *schema.ResourceData, client *opslevel.Client) error {
+	teamId := d.Get("team_id").(string)
+	members := expandStringArray(d.Get("members").(*schema.Set).List())
+
+	team, err := getTeamFromId(teamId, client)
+	if err != nil {
+		return err
+	}
+
+	if err = addMembers(team, members, client); err != nil {
+		return err
+	}
+
+	d.SetId(resource.PrefixedUniqueId(""))
```

Comment:
`resource` is a package in the [terraform plugin sdk](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/resource#PrefixedUniqueId). I'm using the prefixed uniqued id to avoid it assigning an id with `terraform-` in the state, though I don't think that changes anything. It was more of a personal choice 🙈

Reply:
Interesting.  I wonder if this is pointing to an incorrect implementation since team members is not a resource in our API and as such should not be a resource in terraform but rather an array on the `opslevel_team` resource itself?

Reply:
It very well could be, and I'm definitely starting to lean in that direction as well. I'll take a crack at removing the resource and just adding an optional members fields to the team resource 🙏